lib/base-compiler.js

```diff
@@ -844,7 +774,7 @@ class BaseCompiler {
         return this.asm.process(result.asm, filters);
     }
 
-    postProcessAsm(result) {
+    async postProcessAsm(result) {
         if (!result.okToCache || !this.demanglerClass) return result;
         const demangler = new this.demanglerClass(this.compiler.demangler, this);
         return demangler.process(result);
```

Comment:
Great spot! Thanks!

Reply:
Any thoughts on this? My IDE gets grumpy with `return await`...

Reply:
In fairness the IDE gets annoyed here:
![Screenshot from 2019-12-30 15-27-51](https://user-images.githubusercontent.com/633973/71601428-fb16b400-2b18-11ea-9633-6986075438f6.png)

Reply:
It seems my suggestion here is misguided - things work differently in other languages I am more familiar with (namely C#, where it is generally good practice to return await in almost all cases). I think the rules outlined in that eslint page make sense - only return await if you are inside a try/catch block.

As for the other case - you are right. The outer await is required because you are assigning the result of the async function. The inner await is not required because you are returning from the result an async function.

Reply:
Thanks mate!